src/test/scala/com/strad/service/ServiceRoutesSpec.scala

```diff
@@ -0,0 +1,62 @@
+package com.strad.service
+
+import cats.*
+import cats.effect.*
+import cats.implicits.*
+import com.strad.ServiceRoutes
+import com.strad.repositories.UserRepo
+import com.strad.repositories.domain.User
+import com.strad.service.ModelsJson.given
+import com.strad.service.domain.UserResponse
+import io.circe.*
+import io.circe.generic.auto.*
+import io.circe.syntax.*
+import munit.CatsEffectSuite
+import org.http4s.*
+import org.http4s.circe.*
+import org.http4s.circe.CirceEntityDecoder.circeEntityDecoder
+import org.http4s.client.Client
+import org.http4s.dsl.io.*
+import org.http4s.implicits.uri
+
+class ServiceRoutesSpec extends CatsEffectSuite:
+  private val success: UserRepo[IO] = new UserRepo[IO]:
+    override def getUserById(id: Long): IO[Option[User]] = IO.pure(Some(User(123, None)))
+    override def insertUser(user: User): IO[User] = ???
+    override def deleteUserById(userId: Long): IO[Unit] = ???
+
+  private val foundNone: UserRepo[IO] = new UserRepo[IO]:
+    override def getUserById(id: Long): IO[Option[User]] = IO.pure(None)
+    override def insertUser(user: User): IO[User] = ???
+    override def deleteUserById(userId: Long): IO[Unit] = ???
+
+  private val exception: UserRepo[IO] = new UserRepo[IO]:
+    override def getUserById(id: Long): IO[Option[User]] = IO.raiseError(new RuntimeException("Should not get called!"))
+    override def insertUser(user: User): IO[User] = ???
+    override def deleteUserById(userId: Long): IO[Unit] = ???
+
+  private def userServiceRoutes(repo: UserRepo[IO]) =
+    ServiceRoutes.userRoutes[IO](new UserServiceImpl[IO](repo)).orNotFound
```

Comment:
We are trying to test the code as written.  There is ServiceRoutes which define the UserRoutes in the code.  We want to test that without having to connect to a database though.  So we just need to dummy out the repo layer and keep the service layer as it.  You could argue that we should ALSO dummy out the service layer as by having that in there it is testing more than just the routes.  I can get behind that idea tbh and feel free to improve things.